src/lxc/criu.c

```diff
@@ -498,10 +511,21 @@ static void exec_criu(struct criu_opts *opts)
 			case LXC_NET_VETH:
 				veth = n->priv.veth_attr.pair;
 
-				if (n->link)
-					ret = snprintf(buf, sizeof(buf), "veth[%s]:%s@%s", eth, veth, n->link);
-				else
-					ret = snprintf(buf, sizeof(buf), "veth[%s]:%s", eth, veth);
+				if (n->link) {
+					if (external_not_veth)
+						strncpy(fmt, "veth[%s]:%s@%s", sizeof(fmt));
```

Comment:
Aren't we reading potentially invalid memory here? It might be less awkward with just `char *fmt` and then `fmt = "veth[%s]:%s@%s`.